### PR TITLE
Return error when given a non-existent exact version of a system chart

### PR DESCRIPTION
**Pull request by maxsokolovsky**

### CODE REVIEW

**Review by MbolotSuse — APPROVED** [review]
LGTM. I think that this will be difficult to test since this happens async, but I would recommend trying to think of an approach.

**Comment by MbolotSuse** [comment]
Now that this is returning an error, you will need to update: https://github.com/rancher/rancher/blob/c1175854f8e3a96b7fc39dc8110300ced1f6e186/tests/v2/codecoverage/package/Dockerfile as well, otherwise that probably won't be able to find a valid version.

**Review by MbolotSuse — CHANGES_REQUESTED** [review]
Needs to update version used in tests as well.

**Review by KevinJoiner — APPROVED** [review]
LGTM with Michaels comment

File: tests/v2/codecoverage/package/Dockerfile [hunk 1]
```diff
@@ -56,7 +56,7 @@ ENV CATTLE_SYSTEM_UPGRADE_CONTROLLER_CHART_VERSION 101.0.0+up0.3.3
 ENV CATTLE_FLEET_MIN_VERSION=102.0.0+up0.6.0-rc.5
 # Deprecated.
 ENV CATTLE_RANCHER_WEBHOOK_MIN_VERSION=''
-ENV CATTLE_RANCHER_WEBHOOK_VERSION=2.0.2+up0.3.2-rc21
+ENV CATTLE_RANCHER_WEBHOOK_VERSION=2.0.3+up0.3.3
```
Comment (MbolotSuse) [inline tests/v2/codecoverage/package/Dockerfile:59]: Ideally this should be the same RC as the main file is on, but I think that this works for now

**Review by MbolotSuse — APPROVED** [review]

**Review by KevinJoiner — APPROVED** [review]

**Pull request merged**